pkg/cloudprovider/providers/gce/gce_disks.go

```diff
@@ -398,9 +325,12 @@ func (manager *gceServiceManager) getDiskSourceURI(disk *GCEDisk) (string, error
 
 func (manager *gceServiceManager) getDiskTypeURI(
 	diskRegion string, diskZoneInfo zoneType, diskType string) (string, error) {
-	getProjectsAPIEndpoint := manager.getProjectsAPIEndpoint()
+
+	var getProjectsAPIEndpoint string
 	if manager.gce.AlphaFeatureGate.Enabled(AlphaFeatureGCEDisk) {
```

Comment:
Since this method is used by both `CreateDiskOnCloudProvider(...)` and `CreateRegionalDiskOnCloudProvider(...)`, we should probably not key off `AlphaFeatureGate.Enabled(AlphaFeatureGCEDisk)` alone, right? That would force standard disks that are running on alpha cluster to use the alpha api?

Reply:
Agreed, better to pass that info in from the `CreateDisk...` methods.